engine/backends/coq/coq_backend.ml

```diff
@@ -835,21 +837,24 @@ module CoqBackend = struct
                 pty body.typ );
           ]
       | TyAlias { name; generics; ty } ->
-          [ C.AST.Notation (pglobal_ident name ^ "_t", pty ty) ]
+         [ C.AST.Notation (pglobal_ident name ^ "_t", pty ty) ]
       (* record *)
       | Type
           {
             name;
             generics;
             variants = [ { name = record_name; arguments } ];
             record = true;
-          } ->
+          } when name == record_name ->
```

Comment:
Hmm, this is equality between `global_ident`s so I think it might work, but I will do something better.

Reply:
Oh, sorry, indeed!
But then you can just use `GlobalIdent.(name = record_name)`

(otherwise, two equal `global_ident`s that comes from different place in the AST would not be equal `==`-wise)

Reply:
Yes, that is probably better, even though that should never be the case, as these are type declarations?

Reply:
I mean for the specific rust code, we should not have 2 type definitions with the same global identifier, as that would mean you define the same type twice?

Reply:
I seems you cannot do `GlobalIdent.(name = record_name)` but have to do `GlobalIdent.equal name record_name`, as the derive macro produces an `equal` function and not `(=)`?

Reply:
Ok, that's a bit sad the PPX doesn't derive `=`